gravitee-apim-gateway/gravitee-apim-gateway-security/gravitee-apim-gateway-security-core/src/main/java/io/gravitee/gateway/security/core/DefaultAuthenticationHandlerSelector.java

```diff
@@ -33,11 +35,23 @@ public DefaultAuthenticationHandlerSelector(AuthenticationHandlerManager authent
     @Override
     public AuthenticationHandler select(ExecutionContext executionContext) {
         // Prepare the authentication context
-        final SimpleAuthenticationContext context = new SimpleAuthenticationContext(executionContext);
+        final SimpleAuthenticationContext authenticationContext = new SimpleAuthenticationContext(executionContext);
+        final List<AuthenticationHandler> authenticationHandlers = authenticationHandlerManager.getAuthenticationHandlers();
+
+        for (int i = 0; i < authenticationHandlers.size(); i++) {
+            final AuthenticationHandler authenticationHandler = authenticationHandlers.get(i);
+            authenticationContext.setInternalAttribute(ATTR_INTERNAL_LAST_SECURITY_HANDLER_SUPPORTING_SAME_TOKEN_TYPE, true);
+
+            if (i < authenticationHandlers.size() - 1) {
+                // Check whether the next authentication handler works on the same type of token or not.
```

Comment:
Are they sorted by Type ? Shouldn't we test if one of next oneS support the same type?

Reply:
They are sorted by order(). We also discuss this point and trying to change that order by reordering or grouping by type has been quickly abandoned due to possible bad side effects.